packages/@vue/cli/lib/GeneratorAPI.js

```diff
@@ -77,6 +77,24 @@ class GeneratorAPI {
     return this.generator.hasPlugin(id)
   }
 
+  /**
+   * Configure how config files are extracted.
+   *
+   * @param {string} key - Config key in package.json
+   * @param {ConfigTransform} transformer - A function that receives the
+   *   config object, whether to check for an existing config on disk, and the
+   *   current context.
+   */
+  addConfigTransform (key, configTransform) {
```

Comment:
I think the current API would be a bit difficult to use as the plugin author would have to reimplement `makeXXXTransform` or require vue-cli internals (risks breaking).

The second argument could also accept an object in the form of

```
{
  file: '.baberc.js',
  type: 'js|json|multi'
}
```